test/fs/test_fs_js_api.c

```diff
@@ -14,453 +14,444 @@
-
-    stat("/readable", &s);
-    assert(s.st_mode & 0400 /* S_IRUSR */);
-
-    EM_ASM(
-        var ex;
-        try {
-            FS.mkdirTree("/readable/forbidden");
-        } catch (err) {
-            ex = err;
-        }
-        assert(ex.name === "ErrnoError" && ex.errno === 2 /* EACCES */);
-    );
-    
-    remove("/test1/test2/test3");
-    remove("/test1/test2");
-    remove("/test1");
-    remove("/readable");
+  EM_ASM(
+    FS.mkdirTree("/test1/test2/test3");
+
+    FS.mkdirTree("/readable", 0400 /* S_IRUSR */);
+  );
+
+  struct stat s;
+  stat("/test1", &s);
+  assert(S_ISDIR(s.st_mode));
+  stat("/test1/test2", &s);
+  assert(S_ISDIR(s.st_mode));
+  stat("/test1/test2/test3", &s);
+  assert(S_ISDIR(s.st_mode));
+
+  stat("/readable", &s);
+  assert(s.st_mode & 0400 /* S_IRUSR */);
+
+  EM_ASM(
+    var ex;
+    try {
+      FS.mkdirTree("/readable/forbidden");
+    } catch (err) {
+      ex = err;
+    }
+    assert(ex.name === "ErrnoError" && ex.errno === 2 /* EACCES */);
+  );
+
+  remove("/test1/test2/test3");
+  remove("/test1/test2");
+  remove("/test1");
+  remove("/readable");
 }
 
 void test_fs_utime() {
-    EM_ASM(
-        FS.writeFile('utimetest', 'a=1\nb=2\n');
-    );
+  EM_ASM(
+    FS.writeFile('utimetest', 'a=1\nb=2\n');
+  );
 
-    EM_ASM(
-        FS.utime('utimetest', 10500, 8000);
-    );
-    struct stat utimeStats;
-    stat("utimetest", &utimeStats);
+  EM_ASM(
+    FS.utime('utimetest', 10500, 8000);
+  );
+  struct stat utimeStats;
+  stat("utimetest", &utimeStats);
 
-    assert(utimeStats.st_atime == 10);
-    assert(utimeStats.st_atim.tv_sec == 10);
+  assert(utimeStats.st_atime == 10);
+  assert(utimeStats.st_atim.tv_sec == 10);
 
-    assert(utimeStats.st_atim.tv_nsec == 500000000);
-    
-    assert(utimeStats.st_mtime == 8);
-    assert(utimeStats.st_mtim.tv_sec == 8);
+  assert(utimeStats.st_atim.tv_nsec == 500000000);
 
-    remove("utimetest");
-}
+  assert(utimeStats.st_mtime == 8);
+  assert(utimeStats.st_mtim.tv_sec == 8);
 
-void cleanup() {
-    remove("testfile");
-    remove("renametestfile");
-    remove("readtestfile");
-    remove("closetestfile");
+  remove("utimetest");
 }
 
 int main() {
-    test_fs_open();
-    test_fs_createPath();
-    test_fs_readFile();
-    test_fs_rename();
-    test_fs_readlink();
-    test_fs_read();
-    test_fs_rmdir();
-    test_fs_close();
-    test_fs_mknod();
-    test_fs_truncate();
+  test_fs_open();
+  test_fs_createPath();
+  test_fs_readFile();
+  test_fs_rename();
+  test_fs_readlink();
+  test_fs_read();
+  test_fs_rmdir();
+  test_fs_close();
+  test_fs_mknod();
+  test_fs_truncate();
 #if WASMFS
-    // TODO: Fix legacy API FS.mmap bug involving emscripten_builtin_memalign
-    test_fs_mmap();
+  // TODO: Fix legacy API FS.mmap bug involving emscripten_builtin_memalign
+  test_fs_mmap();
 #endif
-    test_fs_mkdirTree();
-    test_fs_utime();
-
-    cleanup();
+  test_fs_mkdirTree();
+  test_fs_utime();
 
```

Comment:
Hmm, I wonder if we need the cleanup for things like IDBFS that persist... but perhaps we don't test this in that mode, which is fine. If we did we'd need to clear it manually.

Reply:
There could be some cases like this, but this is not one of them (IIUC).

In fact its preferable to leave the files around in the test directory afterwards so manual inspection can happen if needed.